ppmclibs/tinycv_impl.cc

```diff
@@ -364,11 +368,34 @@ void create_opencv_threads(int thread_count)
     parallel_for_(Range(0, thread_count), RunFunctionInParallel([&] (const Range &) {
         // keep the thread idling until the expected number of threads has been spawned
         std::unique_lock<std::mutex> lock(m);
+
         if (++threads_spawned >= thread_count) {
             cv.notify_all(); // note: The last thread releases all previously spawned threads.
         } else {
             cv.wait(lock);
         }
+
+        // check whether the signal mask is set as expected
+        auto current_set = sigset_t();
+        sigemptyset(&current_set);
+        auto sigmask_get = pthread_sigmask(SIG_BLOCK, nullptr, &current_set);
+
+        // exit with an error if the signal mask does not contain at least SIGCHLD and SIGTERM
+        if (sigmask_get != 0) {
+            std::cerr << "Unable to get sigprocmask of OpenCV thread: " << std::strerror(errno) << '\n';
+            std::exit(EXIT_FAILURE);
+        }
+        const auto is_sigchld_blocked = sigismember(&current_set, SIGCHLD);
+        const auto is_sigterm_blocked = sigismember(&current_set, SIGTERM);
+        if (is_sigchld_blocked != 1 || is_sigterm_blocked != 1) {
+            if (is_sigchld_blocked == -1 || is_sigterm_blocked == -1) {
+                auto lock = std::unique_lock<std::mutex>(m);
```

Comment:
I guess this it the lock declared on https://github.com/os-autoinst/os-autoinst/pull/2551/files#diff-56ffbf3709dd0405661f7f4e7e2fafcf718a5f766d54f92c31f59c40590048c8L366?

Can you please explain quickly to me this assignment? It is not locking/unlocking the mutex here, or is it?

Reply:
> It is not locking/unlocking the mutex here, or is it?

It *is* locking/unlocking the mutex here using c'tor/d'tor magic. It does that so the log messages are not mixed up.

Reply:
Ah, right. That makes sense. It wasn't clear from the documentation on the first read (as it states the lock is optional, but I see it now). Thanks!